### ISSUE TITLE
IndexedPointInAreaLocator thread safety

### ISSUE THREAD

**Issue by petebankhead** [locationtech/jts#571]
I have code that previously used an `IndexedPointInAreaLocator` to test for many coordinates in a region using multiple threads. This worked fine in 1.16.1 but gives intermittent errors in 1.17.0. If I disable parallelisation in my code around where `locate` is called, all is well again.

I think it traces back to #519 and specifically [this part](https://github.com/dr-jts/jts/blob/40126a5fff7be272e8cc6f0dc8ee084a44c9593f/modules/core/src/main/java/org/locationtech/jts/algorithm/locate/IndexedPointInAreaLocator.java#L81):
```java
public int locate(Coordinate p)
  {
    if (index == null) {
      index = new IntervalIndexedGeometry(geom);
      // no need to hold onto geom
      geom = null;
    }
```
On some rare occasions, I seem to end up with an `IntervalIndexedGeometry` with a `null` geometry inside.

**Comment by dr-jts** [comment]
Yes, that does look like a concurrency problem.  Should be easy to solve by adding `synchronized`.

**Comment by dr-jts** [comment]
Should be fixed by #572.  @petebankhead  can you verify?

**Comment by petebankhead** [comment]
Thanks @dr-jts - yes, that fixes the issue.

Here is a short test that was previously failing:

```java
public void testParallel() {
	   
	   GeometricShapeFactory factory = new GeometricShapeFactory();
	   factory.setSize(100);
	   Polygon polygon = factory.createCircle();
	   
	   int n = 100_000;
	   List<Coordinate> coords = new ArrayList<>();
	   Random rand = new Random(1324);
	   for (int i = 0; i < n; i++) {
		   coords.add(new Coordinate(rand.nextDouble()*100, rand.nextDouble()*100));
	   }
	   
	   long startTime = System.nanoTime();
	   for (int i = 0; i < 1000; i++) {
		   IndexedPointInAreaLocator locator = new IndexedPointInAreaLocator(polygon);
		   coords.parallelStream().forEach(c -> isInside(locator, c));
	   }
	   long endTime = System.nanoTime();
	   double durationMillis = (endTime - startTime)/1000_000.0;
	   System.out.println(String.format("Test time: %.2f ms", durationMillis));
   }
   
   private boolean isInside(IndexedPointInAreaLocator locator, Coordinate coord) {
	   return locator.locate(coord) == Location.INTERIOR;
   }
```

I included timing code because it looks like #572 is calling the synchronized method more than necessary; if line 82 is switched to become
```java
if (index == null)
   createIndex();
```
then I see a substantial speedup (about 5x) in the test.

**Comment by dr-jts** [comment]
That's a significant performance improvement!  I guess `synchronized` calls are still fairly expensive.

Thanks for the suggestion.  Committed in #574.
